denote.el

```diff
@@ -781,43 +781,53 @@ EXTRA-CHARACTERS is an optional string.  See
   str)
 
 (defun denote--slug-hyphenate (str)
-  "Replace spaces and underscores with hyphens in STR.
-Also replace multiple hyphens with a single one and remove any
-leading and trailing hyphen."
-  (replace-regexp-in-string
-   "^-\\|-$" ""
-   (replace-regexp-in-string
-    "-\\{2,\\}" "-"
-    (replace-regexp-in-string "_\\|\s+" "-" str))))
-
-(defun denote--remove-dot-characters (str)
-  "Remove the dot character from STR."
-  (replace-regexp-in-string "\\." "" str))
-
-(defun denote--trim-right-token-characters (str)
-  "Remove =, - and _ from the end of STR."
-  (string-trim-right str "[=_-]+"))
-
-(defun denote--replace-consecutive-token-characters (str)
+  "Replace spaces and underscores with hyphens in STR."
+  (replace-regexp-in-string "_\\|\s+" "-" str))
+
+(defun denote--trim-right-token-characters (str component)
+  "Remove =, - and _ from the end of STR.
+The removal is done only if necessary according to COMPONENT."
+  (cond ((eq component 'title)
+         (string-trim-right str "[=_]+"))
+        ((eq component 'keyword)
+         (string-trim-right str "[=_-]+"))
+        ((eq component 'signature)
+         (string-trim-right str "[_-]+"))))
+
+(defun denote--replace-consecutive-token-characters (str component)
   "Replace consecutive characters with a single one in STR.
-Spaces, underscores and equal signs are replaced with a single
-one in str."
-  (replace-regexp-in-string
-   "-\\{2,\\}" "-"
-   (replace-regexp-in-string
-    "_\\{2,\\}" "_"
-    (replace-regexp-in-string
-     "=\\{2,\\}" "=" str))))
+Hyphens, underscores and equal signs are replaced with a single
+one in str, if necessary according to COMPONENT."
+  (cond (;; -- are allowed in titles
+         (eq component 'title)
+         (replace-regexp-in-string
+          "_\\{2,\\}" "_"
```

Comment:
How do y'all feel about `__+` instead of `_\\{2,\\}`? To me it's easier to read readable.

Reply:
Yes. This is more readable.